File: challenge-load-test-chat/test-by-locust/src/tasks/get_message.py
from locust import task, TaskSet
from endpoints import Endpoints

class GetMessageTask(TaskSet):
    @task
    def get_message(self):
        chat_id = getattr(self.user, 'chat_id', None)
        message_id = getattr(self.user, 'message_id', None)
        if chat_id and message_id:
            self.client.get(Endpoints.MESSAGE_DETAIL_UPDATE.format(chat_id=chat_id, message_id=message_id))

File: challenge-load-test-chat/test-by-k6/src/tasks/get-message.ts
import http from 'k6/http';
import { check } from 'k6';
import { Endpoints } from '../endpoints';

export function getMessage(baseUrl: string, chatId: string, messageId: string) {
  const res = http.get(`${baseUrl}${Endpoints.MESSAGE_DETAIL_UPDATE(chatId, messageId)}`);
  check(res, { 'get message status is 200': (r) => r.status === 200 });
  return res;
}
